docs/understanding-docker.md

```diff
@@ -17,274 +17,287 @@ weight = -90
-basis for a new image, for example if you have a base Apache image you could use
-this as the base of all your web application images.
+An image is defined in a Dockerfile. Every image starts from a base image, such as
+`ubuntu`, a base Ubuntu image, or `fedora`, a base Fedora image. You can also use
+images of your own as the basis for a new image, for example if you have a base
+Apache image you could use this as the base of all your web application images. The
+base image is defined using the `FROM` keyword in the dockerfile.
 
 > **Note:** [Docker Hub](https://hub.docker.com) is a public registry and stores
 images.
 
-Docker images are then built from these base images using a simple, descriptive
-set of steps we call *instructions*. Each instruction creates a new layer in our
-image. Instructions include actions like:
+The docker image is built from the base image using a simple, descriptive
+set of steps we call *instructions*, which are stored in a `Dockerfile`. Each
+instruction creates a new layer in the image. Some examples of Dockerfile
+instructions are:
 
-* Run a command
-* Add a file or directory
-* Create an environment variable
-* What process to run when launching a container from this image
+* Specify the base image (`FROM`)
+* Specify the maintainer (`MAINTAINER`)
+* Run a command (`RUN`)
+* Add a file or directory (`ADD`)
+* Create an environment variable (`ENV`)
+* What process to run when launching a container from this image (`CMD`)
 
-These instructions are stored in a file called a `Dockerfile`. A `Dockerfile` is
-a text based script that contains instructions and commands for building the image
-from the base image. Docker reads this `Dockerfile` when you request a build of
-an image, executes the instructions, and returns a final image.
+Docker reads this `Dockerfile` when you request a build of
+an image, executes the instructions, and returns the image.
 
 ### How does a Docker registry work?
-The Docker registry is the store for your Docker images. Once you build a Docker
-image you can *push* it to a public registry such as [Docker Hub](https://hub.docker.com)
-or to your own registry running behind your firewall.
-
-Using the Docker client, you can search for already published images and then
-pull them down to your Docker host to build containers from them.
-
-[Docker Hub](https://hub.docker.com) provides both public and private storage
-for images. Public storage is searchable and can be downloaded by anyone.
-Private storage is excluded from search results and only you and your users can
-pull images down and use them to build containers. You can [sign up for a storage plan
-here](https://www.docker.com/pricing).
+A Docker registry stores Docker images. After you build a Docker image, you
+can *push* it to a public registry such as [Docker Hub](https://hub.docker.com)
+or to a private registry running behind your firewall. You can also search for
+existing images and pull them from the registry to a host.
+
```

Comment:
It might make sense to mention [Docker Store](https://store.docker.com/beta) here at first mention of Docker Hub. It's currently in Beta but will ultimately (I think) replace Docker Hub. So you might add a line that says something to this effect either here and/or under the Next Steps.